azure/services/scalesetvms/scalesetvms.go

```diff
@@ -151,11 +151,7 @@ func (s *Service) deleteVMSSFlexVM(ctx context.Context, resourceID string) error
 	if err != nil {
 		return errors.Wrap(err, fmt.Sprintf("failed to parse resource id %q", resourceID))
 	}
-
-	resourceGroup := parsed.ResourceGroup
-	resourceName := strings.TrimPrefix(s.Scope.ProviderID(), azure.ProviderIDPrefix)
-	resourceNameSplits := strings.Split(resourceName, "/")
-	resourceName = resourceNameSplits[len(resourceNameSplits)-3] + "_" + resourceNameSplits[len(resourceNameSplits)-1]
+	resourceGroup, resourceName := parsed.ResourceGroup, parsed.ResourceName
```

Comment:
So the response from the autorest library is smart enough to know about flex VM vs vanilla VM? I guess because you're passing in the resource ID we would expect it to be authoritative for parsing the various sub-components that represent the variety of resources across Azure.

Reply:
I think it wasn't before but it now is since VMSS Flex IDs changed? @mboersma please correct me if I'm wrong